src/main/java/io/swagslash/gametrackerserver/igdbconsumer/IGDBWrapper.java

```diff
@@ -110,19 +113,74 @@ public String getMethod() {
     }
 
 
-    public List<Game> searchGames(String searchTerm) {
+    public List<IGDBGame> searchGames(String searchTerm) {
         IGDBQuery query = new IGDBQuery();
         query.setSearch(searchTerm);
         return getGames(query);
     }
 
-    public List<Cover> getCover(Game game) {
+    public IGDBCover getCover(IGDBGame game) {
         return getCover(game.getCover());
     }
 
-    public List<Cover> getCover(Integer coverId) {
+    public IGDBCover getCover(Integer coverId) {
         IGDBQuery query = new IGDBQuery();
         query.setWhere("id=" + coverId);
-        return getCover(query);
+        List<IGDBCover> covers = getCovers(query);
+        assert covers != null;
```

Comment:
Could be `Objects.requireNonNull(covers)`

Should the application really crash if loading fails?

Reply:
`Objects.requireNonNull` will still throw a null pointer, just explicitly right? So I don't see the benefits.
But yes, the assert is probably overkill.